update example 39-mcp-stdio.py to use different mcp server

vipyne opened the pull request:
https://www.loom.com/share/a9f0a270261d4c6cb054ab2b4dcd6084

SO to Rijksmuseum MCP
https://github.com/r-huijts/rijksmuseum-mcp

#### Please describe the changes in your PR. If it is addressing an issue, please reference that as well.

closes #2965

markbackman commented:
What was the issue with the NASA MCP server? I noticed it was returning a BrokenResourceError. Should the MCPClient handle that type of error more gracefully or was it just a one-off issue that we are confident is related to the NASA MCP server?

vipyne commented:
> What was the issue with the NASA MCP server? 

`anyio.BrokenResourceError`
My wild guess is there is a conflict between `anyio` (used by NASA MCP server) and `asyncio`.
the error is thrown after this line `tools = await mcp.register_tools(llm)`
I added a print statement [here](https://github.com/pipecat-ai/pipecat/blob/main/src/pipecat/services/mcp_service.py#L233) and tools_schema prints, so my thought is that when that session closes (after line 233) the NASA MCP server thinks we have disconnected and it throws `BrokenResourceError` because according to AnyIO code:
```
:raises ~anyio.BrokenResourceError: if the stream has been closed from the receiving end
```

> Should the MCPClient handle that type of error more gracefully

if your github PAT token is invalid in example `39c-mcp-run-http.py`, a similar error stack is thrown:
```
...
| Traceback (most recent call last):
|   File "/Users/.../pipecat-ai/pipecat/.venv/lib/python3.12/site-packages/mcp/client/streamable_http.py", line 405, in handle_request_async
|     await self._handle_post_request(ctx)
|   File "/Users/.../pipecat-ai/pipecat/.venv/lib/python3.12/site-packages/mcp/client/streamable_http.py", line 277, in _handle_post_request
|     response.raise_for_status()
|   File "/Users/.../pipecat-ai/pipecat/.venv/lib/python3.12/site-packages/httpx/_models.py", line 829, in raise_for_status
|     raise HTTPStatusError(message, request=request, response=self)
| httpx.HTTPStatusError: Client error '401 Unauthorized' for url 'https://api.githubcopilot.com/mcp/'
| For more information check: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401
```
with a `httpx.HTTPStatusError` instead of `anyio.BrokenResourceError`.
This and the nasa mcp error can be caught/handled by wrapping the `tools = await mcp.register_tools(llm)` line in a try.  I'll go ahead add that try in the examples in this PR.

If we want to try to handle that within the mcp_service class, we can do that, but I'd argue it's out of scope in this PR.  Happy to make an issue to handle this type of error within the mcp_service `register_tools` function.

> or was it just a one-off issue that we are confident is related to the NASA MCP server?

it seems to be a quirk of this specific server mixed with pipecat.  at one point it worked.  but the server I added in this PR, `rijksmuseum-mcp` works. `"npm-search-mcp-server"`* works. 

I will continue to look into the issue with nasa mcp (according to the mcp inspector, it's not broken per say, but there are a few broken resource links so... I'd hesitate to spend a ton of time making this particular (unofficial) mcp server working.  

in the meantime, this fixes the example (and I think it is actually a more impressive example anyway).

*try with:
```
mcp = MCPClient(
    server_params=StdioServerParameters(
        command=shutil.which("npx"),
        args=["-y", "npm-search-mcp-server"],
    )
)
```

markbackman commented:
> If we want to try to handle that within the mcp_service class, we can do that, but I'd argue it's out of scope in this PR.

Agreed. Just wanted to make sure it wasn't something universal. Thanks for fixing this 🙏
